### PR TITLE
a few search and sort enhancements

### LINKED ISSUE #314: Community pages showing all posts
**Describe the bug**
Under the community pages, under activity, all posts are showing,

**To Reproduce**
1. Go to any community pages [for example](https://near.social/devgovgigs.near/widget/gigs-board.pages.community.activity?handle=component-libraries)
2. Go to activity, look in the list
3. See all posts

**Expected behaviour**
Only posts made under the community or tagged with the community should be shown here 

**Screenshots**
image
Component libraries but near campus is showing

**Desktop (please complete the following information):**
OS: Windows 11
Browser: Chrome

### LINKED ISSUE #324: Display all posts including replies
**Is your feature request related to a problem? Please describe.**
It is hard to see of activity happening on DevHub now that there is no option to see all replies.

**Describe the solution you'd like**
A button that shows all replies.

**Describe alternatives you've considered**
I couldn't come up with a better alternative.

### PR DESCRIPTION
**Pull request by ailisp**
Fix #324, Fix #314 
Also make search instantly trigger when typing, instead of pressing enter to search.

Preview: https://near.social/bo.near/widget/gigs-board.pages.Feed?nearDevGovGigsContractAccountId=devgovgigs.near

Note:
- Search by default seaching only posts, not comments
- Select "Sort -> All replies" it will list latest comments and posts
- After "Sort -> All replies" then search will search all comments and posts
- Click on a community now correctly list posts of that community

### CODE REVIEW

**Review by frol — CHANGES_REQUESTED** [review]
> After "Sort -> All replies" then search will search all comments and posts

This is not really intuitive since it is not "sorting". I guess you may want to rename it to "filter" and default would be "Top-level posts".

There is also a bug: the state of this dropdown is not displayed after selection.

**Comment by frol** [comment]
Another bug related to #314 - tags filtering does not work when you click on a tag of a post (see #226) it was previously resolved, but it is an issue again.

**Comment by ailisp** [comment]
@frol 

> This is not really intuitive since it is not "sorting". I guess you may want to rename it to "filter" and default would be "Top-level posts".

I have the same feeling here, your proposed way is more intuitive to me, but this is trying to match original design: 
image

(Sorting by hottest is WIP in indexer side). Let's come back to consider it after getting it matches to the design.

I'll fix these ones:

> There is also a bug: the state of this dropdown is not displayed after selection. 
> Another bug related to https://github.com/near/neardevhub-widgets/issues/314 - tags filtering does not work when you click on a tag of a post (see https://github.com/near/neardevhub-widgets/issues/226) it was previously resolved, but it is an issue again.

File: src/gigs-board/feature/post-search/panel.jsx [hunk 1]
```diff
@@ -63,6 +67,8 @@ const buttonStyle = {
   color: "#f3f3f3",
 };
 
+console.log(state.tag);
```
Comment (frol) [inline src/gigs-board/feature/post-search/panel.jsx:70]: Remove the debug logs

**Review by frol — APPROVED** [review]
All the reported issues are addressed and this PR looks good to me

**Pull request merged**